pyaedt/edb_core/EDB_Data.py

```diff
@@ -13,6 +16,431 @@
+        """
+        Get the points to be plot
+        """
+        # fmt: off
+        x = []
+        y = []
+        for i, point in enumerate(my_net_points):
+            # point = my_net_points[i]
+            if not point.IsArc():
+                x.append(point.X.ToDouble())
+                y.append(point.Y.ToDouble())
+                # i += 1
+            else:
+                arc_h = point.GetArcHeight().ToDouble()
+                p1 = [my_net_points[i-1].X.ToDouble(), my_net_points[i-1].Y.ToDouble()]
+                if i+1 < len(my_net_points):
+                    p2 = [my_net_points[i+1].X.ToDouble(), my_net_points[i+1].Y.ToDouble()]
+                else:
+                    p2 = [my_net_points[0].X.ToDouble(), my_net_points[0].Y.ToDouble()]
+                x_arc, y_arc = self._eval_arc_points(p1, p2, arc_h, num)
+                x.extend(x_arc)
+                y.extend(y_arc)
+                # i += 1
+        # fmt: on
+        return x, y
+
+    @aedt_exception_handler
+    def points(self, arc_segments=6):
+        """Return the list of points with arcs converted to segments.
+
+        Parameters
+        ----------
+        arc_segments : int
+            Number of facets to convert an arc. Default is `6`.
+
+
+        Returns
+        -------
+        list, list
+            x and y list of points.
+        """
+        try:
+            my_net_points = list(self.primitive_object.GetPolygonData().Points)
+            xt, yt = self._get_points_for_plot(my_net_points, arc_segments)
+            if not xt:
+                return []
+            x, y = GeometryOperators.orient_polygon(xt, yt, clockwise=True)
+            return x, y
+        except:
+            x = []
+            y = []
+        return x, y
+
+    @property
+    def voids(self):
+        """Return a list of voids of the given primitive if any.
+
+        Returns
+        -------
+        list of :class:`pyaedt.edb_core.EDB_Data.EDBPrimitives`
+        """
+        voids = []
+        for void in self.primitive_object.Voids:
+            voids.append(EDBPrimitives(void, self._app))
+        return voids
+
+    @aedt_exception_handler
+    def points_with_arcs(self):
+        """Return a list of points in [x,y] coordinate.
+        For arc, y is infinite and the arc height is represented with x coordinate.
+
+        Returns
+        -------
+        list of list
+            list of [x,y] couple of data.
+        """
+        points = []
+        try:
+            my_net_points = list(self.primitive_object.GetPolygonData().Points)
+            for i, point in enumerate(my_net_points):
+                points.append([point.X.ToDouble(), point.Y.ToDouble()])
+            return points
+        except:
+            points = []
+        return points
+
+    @aedt_exception_handler
+    def is_arc(self, point):
+        """Either if a point is an arc or not.
+
+        Returns
+        -------
+        bool
+        """
+        return not point.IsArc()
```

Comment:
Should it be:
```suggestion
        return point.IsArc()
```
It is surprising to have the ``not``. Maybe the `IsArc()` implementation is not intuitive.